src/api/spec/features/beta/webui/maintenance_workflow_spec.rb

```diff
@@ -0,0 +1,126 @@
+  end
+
+  it 'maintenance workflow without patchinfo' do
+    # # Step 1: The user branches a package
+    # ####################################
+    login(user)
+
+    # visit package_show_path(project: update_project, package: maintenance_package)
+    # desktop? ? click_link('Branch Package') : click_menu_link('Actions', 'Branch Package')
+    # expect(page).to have_text('Source')
+
+    # click_button('Branch')
+
+    # expect(page).to have_text('Successfully branched package')
+
+    target_packages = []
+    user.run_as do
+      target_packages << BranchPackage.new(project: project.name, package: maintenance_package.name, target_project: 'home:tom:branches:Update',
+                                           target_package: 'cacti.openSUSE:11.4').branch
+    end
+    user.run_as do
+      target_packages << BranchPackage.new(project: another_project.name, package: another_package.name, target_project: 'home:tom:branches:Update',
+                                           target_package: 'cacti.openSUSE:11.5').branch
+    end
+    update_project_branch = Project.last
+    target_packages.each do |package|
+      Backend::Connection.put("/source/#{package[:data][:targetproject]}/#{package[:data][:targetpackage]}/DUMMY_FILE", 'dummy')
+    end
+
+    # Create the patchinfo
+    visit project_show_path(project: Project.last.name)
+    desktop? ? click_link('Create Patchinfo') : click_menu_link('Actions', 'Create Patchinfo')
+    fill_in('patchinfo_summary', with: 'ProjectWithRepo_package is much better than the old one')
+    fill_in('patchinfo_description', with: 'Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing, Fixes nothing')
+    check('patchinfo_block')
+    fill_in('patchinfo_block_reason', with: 'locked!')
+    click_button('Save')
+    expect(page).to have_css('#flash', text: 'Successfully edited patchinfo')
+
+    # Step 2: The user submits the update creating a maintenance incident
+    #####################################################################
+    visit project_show_path(project: Project.last.name)
+
+    expect(page).to have_text('patchinfo')
+
+    desktop? ? click_link('Submit as Update') : click_menu_link('Actions', 'Submit as Update')
+    expect(page).to have_title('Submit as Update')
+    fill_in('description', with: 'I want the update')
+    save_screenshot('about to submit the update.png')
+    click_button('Submit')
+
+    expect(page).to have_text('Created maintenance incident request')
+
+    logout
+
+    # Step 3: The maintenance coordinator accepts the request
+    #########################################################
+    login(maintenance_coord_user)
+
+    visit request_show_path(bs_request)
+    save_screenshot('maintenance_request.png')
+    expect(page).to have_text('This is a Maintenance Incident')
+    fill_in('reason', with: 'really? ok')
+
+    click_button('Accept request')
+    expect(page).to have_text("Request #{bs_request.number} accepted")
```

Comment:
This is another context or even another spec.

- https://www.betterspecs.org/#contexts